singlestoredb/management/job.py

```diff
@@ -0,0 +1,792 @@
+                target_config['targetType'] = TargetType.WORKSPACE.value
+
+            elif cluster_id is not None:
+                target_config['targetID'] = cluster_id
+                target_config['targetType'] = TargetType.CLUSTER.value
+
+        job_run_json = dict(
+            schedule=schedule,
+            executionConfig=execution_config,
+        )  # type: Dict[str, Any]
+
+        if target_config is not None:
+            job_run_json['targetConfig'] = target_config
+
+        if name is not None:
+            job_run_json['name'] = name
+
+        if description is not None:
+            job_run_json['description'] = description
+
+        res = self._manager._post('jobs', json=job_run_json).json()
+        return Job.from_dict(res, self)
+
+    def run(
+        self,
+        notebook_path: str,
+        runtime_name: Optional[str] = None,
+        parameters: Optional[List[Tuple[str, Any]]] = None,
+    ) -> Job:
+        """
+        Creates and returns a scheduled notebook job that
+        runs once immediately on a specific runtime
+        With
+        """
+        return self.schedule(
+            notebook_path,
+            Mode.ONCE,
+            False,
+            start_at=datetime.datetime.now(),
+            runtime_name=runtime_name,
+            parameters=parameters,
+        )
+
+    def wait(self, jobs: List[Union[str, Job]], timeout: Optional[int] = None) -> bool:
+        if timeout is not None:
+            if timeout <= 0:
+                return False
+            finish_time = datetime.datetime.now() + datetime.timedelta(seconds=timeout)
+
+        for job in jobs:
+            if timeout is not None:
+                job_timeout = int((finish_time - datetime.datetime.now()).total_seconds())
+            else:
+                job_timeout = None
+
+            res = self._wait_for_job(job, job_timeout)
+            if not res:
+                return False
+
+        return True
+
+    def _wait_for_job(self, job: Union[str, Job], timeout: Optional[int] = None) -> bool:
+        if self._manager is None:
+            raise ManagementError(msg='JobsManager not initialized')
+
+        if timeout is not None:
+            if timeout <= 0:
+                return False
+            finish_time = datetime.datetime.now() + datetime.timedelta(seconds=timeout)
+
+        if isinstance(job, str):
+            job_id = job
+        else:
+            job_id = job.job_id
+
+        while True:
+            if timeout is not None and datetime.datetime.now() >= finish_time:
+                return False
+
+            res = self._manager._get(f'jobs/{job_id}').json()
+            job = Job.from_dict(res, self)
+            if job.schedule.mode == Mode.ONCE and job.completed_executions_count > 0:
+                return True
+            if job.schedule.mode == Mode.RECURRING:
+                raise ValueError(f'Cannot wait for recurring job {job_id}')
+            time.sleep(1)
```

Comment:
Does this need to be this short of an interval? 1 second intervals hitting the server is pretty fast when jobs likely take much longer than that to run. If you think we need that responsive of a result, that's fine. I just wanted to check.

Reply:
Changed to 5s as aligned with @ricardoasmarques